build_tools/cmake/iree_cross_compile.cmake

```diff
@@ -0,0 +1,219 @@
+    set(_RULE_CONFIG "$<CONFIG>")
+  endif()
+  if (CMAKE_GENERATOR MATCHES "Make")
+    # Use special command for Makefiles to support parallelism.
+    set(${cmd_var} "$(MAKE)" "-C" "${bin_dir}" "${target}" PARENT_SCOPE)
+  else()
+    set(${cmd_var} "${CMAKE_COMMAND}" --build ${bin_dir} --target ${target} --config ${_RULE_CONFIG} PARENT_SCOPE)
+  endif()
+endfunction()
+
+# iree_host_install
+#
+# Defines custom commands and targets for installing the given `target` under
+# host configuration. The custom target for install will be named as
+# `iree_host_install_${target}`.
+#
+# Precondition:
+# - iree_create_configuration(HOST) is invoked previously.
+#
+# Parameters:
+# - COMPONENT: installation component; used for filtering installation targets.
+# - PREFIX: the root installation path prefix.
+# - DEPENDS: addtional dependencies for the installation.
+function(iree_host_install target)
+  cmake_parse_arguments(_RULE "" "COMPONENT;PREFIX" "DEPENDS" ${ARGN})
+  if(_RULE_COMPONENT)
+    set(component_option -DCMAKE_INSTALL_COMPONENT="${_RULE_COMPONENT}")
+  endif()
+  if(_RULE_PREFIX)
+    set(prefix_option -DCMAKE_INSTALL_PREFIX="${_RULE_PREFIX}")
+  endif()
+
+  iree_get_executable_path(${target} output_path)
+
+  add_custom_command(
+    OUTPUT ${output_path}
+    DEPENDS ${_RULE_DEPENDS}
+    COMMAND "${CMAKE_COMMAND}" ${component_option} ${prefix_option}
+            -P "${IREE_HOST_BINARY_ROOT}/cmake_install.cmake"
+    USES_TERMINAL)
+
+  # Give it a custom target so we can drive the generation manually
+  # when useful.
+  add_custom_target(iree_host_install_${target} DEPENDS ${output_path})
+endfunction()
+
+# iree_declare_host_excutable
+#
+# Generates custom commands and targets for building and installing a tool on
+# host for cross-compilation.
+#
+# Precondition:
+# - iree_create_configuration(HOST) is invoked previously.
+#
+# Parameters:
+# - target: the target to build on host.
+# - BUILDONLY: only generates commands for building the target.
+# - DEPENDS: any additional dependencies for the target.
+function(iree_declare_host_excutable target)
+  cmake_parse_arguments(_RULE "BUILDONLY" "" "DEPENDS" ${ARGN})
+
+  iree_get_executable_path(${target} output_path)
+
+  iree_get_build_command(${target} ${IREE_HOST_BINARY_ROOT} build_cmd)
+
+  add_custom_target(iree_host_build_${target}
+                    COMMAND ${build_cmd}
+                    DEPENDS iree_configure_HOST ${_RULE_DEPENDS}
+                    WORKING_DIRECTORY "${IREE_HOST_BINARY_ROOT}"
+                    COMMENT "Building host ${target}..."
+                    USES_TERMINAL)
+
+  if(_RULE_BUILDONLY)
+    return()
+  endif()
+
+  iree_host_install(${target}
+                    COMPONENT ${target}
+                    PREFIX ${IREE_HOST_BINARY_ROOT}
+                    DEPENDS iree_host_build_${target})
+
+  # Give it a custom target so we can drive the generation manually
+  # when useful.
+  add_custom_target(iree_host_${target} DEPENDS "${output_path}")
```

Comment:
Don't we want this even if it's BUILDONLY?

Reply:
This is actually the installation target  "reference" so it should be gated by `NOT BUILDONLY`. Made it more clear in the comment.